support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionDirectPurchaseBuilder.scala

```diff
@@ -32,11 +33,11 @@ class DigitalSubscriptionDirectPurchaseBuilder(
     val contractAcceptanceDate = todaysDate.plusDays(config.defaultFreeTrialPeriod + config.paymentGracePeriod)
 
     val subscriptionData = subscribeItemBuilder.buildProductSubscription(
-      productRatePlanId,
-      overridePricingIfRequired(state.product, acquisitionData.map(_.supportAbTests)),
+      productRatePlanId = productRatePlanId,
+      ratePlanCharges = overridePricingIfRequired(state.product, acquisitionData.map(_.supportAbTests)),
       contractEffectiveDate = todaysDate,
       contractAcceptanceDate = contractAcceptanceDate,
-      readerType = state.product.readerType,
+      readerType = overrideReaderTypeIfRequired(state.promoCode, state.product.readerType),
```

Comment:
would it be possible set this up front so it's consistent, rather than overriding it when we write to zuora? or would that cause other issues?

Reply:
I get your point, however there is no way to be certain that the promo code is validated - only the backend can do that - so it would be risky for a promo code to drive any changes to the reader type at this stage of the process.

Reply:
agree, makes sense to validate before writing to zuora.